### ISSUE TITLE
rsyslog 8.1907.0 omfile/imudp issues

### ISSUE THREAD

**Issue by Fazik** [rsyslog/rsyslog#3748]
### Expected behavior
Logs received continuesly
### Actual behavior
Rsyslog stops writing logs after some period of time
### Steps to reproduce the behavior
We have next setup:
Some app instances writing logs. These logs picked up by rsyslog on them and forwarded to "central" rsyslog instance.
There are something like 70-80 such app instances
"Central" rsyslog instances stores incoming logs as files and also sending this stored logs to 3rd party via tcp+gtls
Currently we are observing issues when "central" rsyslog will stop writing logs to files after some period 
### Environment
- rsyslog version: 8.1907.0
- platform: Ubuntu 18.04 EC2 instance

"Central" rsyslog configuration:

```
main_queue(
           queue.size="2000000"
           queue.workerthreads="1"
           queue.type="Direct"
           queue.fileName="rsyslog-main.spool"
           queue.saveOnShutdown="on"
           queue.maxFileSize="10g"
           queue.highwatermark="2000000"
           queue.lowwatermark="2000000"
           queue.discardMark="2000000"
           queue.workerthreadminimummessages="200"
           queue.dequeuebatchsize="2000"
           )
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat
$RepeatedMsgReduction on
global (
 workDirectory="/opt/od/logs/spool"
 defaultNetstreamDriver="gtls"  # use gtls netstream driver
 defaultNetstreamDriverCAFile="/etc/ssl/rsyslog/ca.pem"  # certificate files - just CA for a client
 defaultnetstreamdrivercertfile="/etc/ssl/rsyslog/cert.pem"
 defaultnetstreamdriverkeyfile="/etc/sssl/rsyslog/key.pem"
)
module(load="builtin:omfile" fileOwner="syslog" fileGroup="adm" fileCreateMode="0644" dirCreateMode="0755")
module(load="imuxsock")
module(load="imklog")
module(load="imfile" PollingInterval="10")
module(load="imrelp")
module(load="imudp")
module(load="imtcp" StreamDriver.Name="gtls" StreamDriver.AuthMode="x509/name" PermittedPeer="*.eu-west-1.compute.internal" StreamDriver.Mode="1")
input(type="imrelp" port="20514")
input(type="imudp" port="514")
input(type="imtcp" port="10514")
$EscapeControlCharactersOnReceive off
$RepeatedMsgReduction off
$ActionResumeInterval 1
$Umask 0022
$PrivDropToUser root
$PrivDropToGroup root
template(name="auth_logger" type="string" string="/logs/%hostname:5:$%/auth.log")
template(name="siem_logger" type="string" string="/logs/%hostname:R,ERE,1,FIELD:^(.+)_.*--end%/%hostname%.log")
template(name="waf_logger" type="string" string="/logs/%hostname:0:3%/waf.log")
template(name="snort_logger" type="string" string="/logs/%hostname:16:$%/alert.log")

if ($hostname startswith "waf_to_siem")       then {action(name="waf" type="omfile" DynaFile="waf_logger" queue.type="Direct")stop}
if ($hostname startswith "snort_siem")    then {action(name="snort" type="omfile" DynaFile="snort_logger" queue.type="Direct")stop}
if ($hostname startswith "auth-")              then {action(name="auth" type="omfile" DynaFile="auth_logger" queue.type="Direct")stop}
if ($hostname contains   "_siem")             then {action(name="siem" type="omfile" DynaFile="siem_logger" queue.type="Direct")stop}
if ($hostname startswith "nat")               then {stop}

```

**Comment by Fazik** [comment]
Sorry for slightly wrong information.
Looks like issue somewhere with imudp
As logs that are coming to 10514 and 20514 are fine

**Comment by davidelang** [comment]
see the topic "8.1905.0 Logging Stops" we thought we had it solved for 8.1907, 
but there is one other report questioning this, look at that issue and see if it 
matches your config, and see if you can produce a debug report

David Lang

**Comment by Fazik** [comment]
Sorry for the late reply.
After upgrade to 1908 everything is fine
